codec_union.go

```diff
@@ -275,6 +296,56 @@ func (d *unionNullableDecoder) Decode(ptr unsafe.Pointer, r *Reader) {
 	d.decoder.Decode(*((*unsafe.Pointer)(ptr)), r)
 }
 
+func encoderOfUnionConverterCodec(_ *encoderContext, schema Schema, typ reflect2.Type) ValEncoder {
+	union := schema.(*UnionSchema)
+	var nullIdx int32
+	var nullable bool
+
+	for i, unionSchema := range union.Types() {
+		if unionSchema.Type() == Null {
+			nullIdx = int32(i)
+			nullable = true
+		}
+	}
+
+	return &unionConverterToAnyCodec{
+		schema:   union,
+		typ:      typ,
+		nullable: nullable,
+		nullIdx:  nullIdx,
+	}
+}
+
+type unionConverterToAnyCodec struct {
+	schema   *UnionSchema
+	typ      reflect2.Type
+	nullable bool
+	nullIdx  int32
+}
+
+func (e *unionConverterToAnyCodec) Encode(ptr unsafe.Pointer, w *Writer) {
+	if *((*unsafe.Pointer)(ptr)) == nil {
+		if !e.nullable {
+			w.Error = errors.New("avro: unionConverterToAnyCodec: encoding nil value for non nillable union")
+			return
+		}
+		w.WriteInt(e.nullIdx)
+		return
+	}
+
+	target := e.typ.UnsafeIndirect(ptr)
+	marshaller := target.(UnionConverter)
+	val, err := marshaller.ToAny()
+	if err != nil {
+		w.Error = err
```

Comment:
I suggest you give context to the error, to make it more clear.
```suggestion
		w.Error = fmt.Errorf("avro: unable to convert union: %w", err)
```